### PR TITLE
feat: ✨ NavigationTree

### PR DESCRIPTION
**Pull request by Roaa94**
* Adds `NavigationTree`  
* `NavigationTree`  & `NavigationTreeNode` added to `widgetbook_for_widgetbook`
* Major changes on `NavigationTreeItem`
* 🧪 Added and updated tests

### CODE REVIEW

File: packages/widgetbook_core/lib/src/navigation_tree/models/navigation_node.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:widgetbook_core/src/navigation_tree/enums/navigation_node_type.dart';
+
+part 'navigation_node.freezed.dart';
+
+@freezed
+class NavigationNode with _$NavigationNode {
```
Comment (Roaa94) [inline packages/widgetbook_core/lib/src/navigation_tree/models/navigation_node.dart:8]: I think the class names in general in this PR might be improved. What do you think? @jenshor

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            offset: Offset(0, isExpanded ? 0 : -1),
+            child: AnimatedAlign(
+              duration: const Duration(milliseconds: 200),
+              curve: Curves.easeInOut,
+              alignment: Alignment.bottomCenter,
+              heightFactor: isExpanded ? 1 : 0,
+              child: ListView.builder(
+                physics: const NeverScrollableScrollPhysics(),
+                itemCount: widget.node.children.length,
+                shrinkWrap: true,
```
Comment (jenshor) [inline packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart:64]: Love the animation 🎉
In combination with the animated expander icon it looks really smooth

**Review by jenshor — COMMENTED** [review]
LGTM. Let me know what you think about my comments 👍🏻

File: packages/widgetbook_core/lib/src/navigation_tree/models/navigation_node.dart (hunk 1 above)
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/navigation_tree/models/navigation_node.dart:8]: That's a great idea. I usually do the `data` suffix in these cases but I hesitated here because of the length, but I agree that consistency is more important. Will make the change

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+class NavigationTree extends StatefulWidget {
+  const NavigationTree({
+    super.key,
+    required this.nodes,
+    this.onNodeSelected,
+  });
+
+  final List<NavigationNode> nodes;
+  final ValueChanged<String>? onNodeSelected;
```
[the remark this answers is not in the document]
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree.dart:12]: My idea was that only the `id` will need to be returned from the `NavigationTree` and the `id` will only have a value if the node is a use case (or a selectable node in general), so I'm calling this method only when the `id` is not null.
But I can do the same logic for the entire node data. Do you prefer this or that I change `onNodeSelected` to `onNodeIdSelected`?

[the remark this answers is not in the document]
Comment (jenshor) [reply packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree.dart:12]: Ok got it. I still think `onNodeSelected` makes sense (and is also more re-usable). Based on that we should return a `NavigationTreeNodeData` object.

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_item.dart [hunk 4]
```diff
@@ -98,70 +31,82 @@ class _NavigationTreeItemState extends State<NavigationTreeItem> {
...
-            Expanded(
-              child: widget.content ??
-                  Text(
-                    widget.label!,
-                    maxLines: 1,
-                    overflow: TextOverflow.ellipsis,
+        child: ClipRRect(
+          borderRadius: BorderRadius.circular(NavigationTreeItem.iconSize),
+          child: ColoredBox(
+            color: isSelected ? const Color(0xff36485F) : Colors.transparent,
```
[the remark this answers is not in the document]
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_item.dart]: Added to the theme 👍

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree.dart]: Updated 👍

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  const NavigationTreeNode({
+    super.key,
+    required this.node,
+    this.level = 0,
+    this.initiallyExpanded = true,
+    this.selectedNodeId,
+    this.onNodeSelected,
+  });
+
+  final NavigationTreeNodeData node;
```
Comment (jenshor) [inline packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart:14]: maybe rename this to data?

File: packages/widgetbook_core/lib/src/themes/themes.dart [hunk 6]
```diff
@@ -43,7 +43,7 @@ class Themes {
     secondary: Color(0xFFBBC7DB),
     onSecondary: Color(0xFF263141),
     secondaryContainer: Color(0xFF3C4858),
-    onSecondaryContainer: Color(0xFFD7E3F8),
+    onSecondaryContainer: Color(0xff36485F),
```
Comment (jenshor) [inline packages/widgetbook_core/lib/src/themes/themes.dart:46]: Why was this changed?

**Review by jenshor — COMMENTED** [review]
I like the new names better. Found two more things. Let me know 😊

File: packages/widgetbook_core/lib/src/themes/themes.dart (hunk 6 above)
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/themes/themes.dart:46]: This is related to this:
https://github.com/widgetbook/widgetbook/pull/415#discussion_r1058173179

And this value now matches the Figma design.
I think in general we should work on the themes such that all colors match the new design.

File: packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart (hunk 5 above)
Comment (Roaa94) [reply packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_node.dart]: Updated

**Review by jenshor — APPROVED** [review]
LGTM 🎉
